packages/taquito/src/contract/rpc-contract-provider.ts

```diff
@@ -764,6 +765,75 @@ export class RpcContractProvider extends Provider implements ContractProvider, S
 
     return batch;
   }
+
+  /**
+   *
+   * @description Stake founds. Will sign and inject an operation using the current context
+   *
+   * @returns An operation handle with the result from the rpc node
+   *
+   * @param stake operation parameter
+   */
+  async stake(params: StakingParams) {
+    const estimate = await this.estimate(params, this.estimator.stake.bind(this.estimator));
+    const source = await this.signer.publicKeyHash();
+
+    const prepared = await this.prepare.stake({ ...params, ...estimate });
+    const content = prepared.opOb.contents.find(
+      (op) => op.kind === OpKind.TRANSACTION
+    ) as OperationContentsTransaction;
+
+    const opBytes = await this.forge(prepared);
```

Comment:
It wasn't mentioned in this ticket, think it's worth adding cases of stake, unstake and finalized_unstake in `integration-tests/data/allTestsCases.ts` to test our forger.forge() function in `integration-tests/local-forging.spec.ts`

Reply:
bump

Reply:
Yes. I did this in the next commit. Can you check please?